### ISSUE TITLE
Strange basic auth behavior when authenticating different users (for a base URL with port 80)

### ISSUE THREAD

**Issue by schmist** [cypress-io/cypress#5185]
### Current behavior:

During tests we are switching users as stated below. It all worked perfectly until we changed the backend port (static routes, API calls) to **80**. Now, **sometimes** Cypress fails to send the new basic auth headers to the backend when changing to another user.

When using other ports except `80` or `443` everything works fine, **always**. Also for other privileged ports. 

Are we using Cypress correctly? Are there other options to authenticate another user? Or is it possible to clear the internal authentication state?

### Desired behavior:

Switching to another user (using `cy.visit` and `auth`) makes Cypress send the currently valid basic auth headers, even on **port 80** and **always**.

### Steps to reproduce: (app code and test code)

```js
describe('some test', () => {
  cy.visit(path, { auth: {username: user, password: 'pw'} as Auth })
  ...some test code...
});
describe('some other test', () => {
  cy.visit(path, { auth: {username: other-user, password: 'other-pw'} as Auth })
  ...test fails, wrong user
});
```

### Versions

Cypress 3.4.0
Linux 5.0.0-27
Electron 61

**Comment by ryan-snyder** [comment]
First off, from the cypress docs https://docs.cypress.io/api/commands/visit.html#Add-basic-auth-headers you should be passing in the options as `cy.visit(path, { auth: { username: user, password: 'pw} as Auth}` That being said, if you pass in the auth option, it should always send whatever you pass in at lest from my understanding. Would be interesting to see what happens if you pass in the auth options the proper way though

**Comment by schmist** [comment]
Sorry, I forgot the curly brackets. We are using the syntax you posted. I updated the issue.

**Comment by ryan-snyder** [comment]
oh ok. Like I said then, I don't see any reason why cypress wouldn't send what you pass in to visit. Do you run your tests on a CI? If so do they fail there and locally? Or just locally?

**Comment by jennifer-shehane** [comment]
@schmist Have you verified that the auth headers work properly outside of Cypress? 

Unfortunately we'll have to close this issue if no reproducible example is provided though. There are a lot of factors involved in what you're describing alone. Can you provide a way for us to reproduce this?

**Comment by schmist** [comment]
@jennifer-shehane Yes we are pretty sure that everything works as it should outside of Cypress. And it used to work also for Cypress until we switched one of our environment to port 80 (nothing else changed).

I will try to provide some example code, but it might take a while.

But are we using Cypress correctly? Should it be possible to switch to a different authenticated user the way we are doing it?

@ryan-snyder It fails on different machines, CI and on local developer workstations.

**Comment by schmist** [comment]
Here is a minimal example:

Run this with node (10.x). It will print the received basic auth headers.
```
const http = require('http'):

http.createServer(function (request, response) {
    console.log(request.headers.authorization);
    response.writeHead(200, {'Content-Type': 'text/html'});
    response.write('foo');
    response.end();
}).listen(80);
```
Use the following Cypress config:
```
{
  "baseUrl": "http://localhost:80"
}

```

Then run the following test:
```
describe('visit server and changes authenticated user', function () {
    it('changes user', function () {
        cy.visit('/', {auth: {username: 'foo', password: 'foo'}});
        cy.visit('/', {auth: {username: 'bar', password: 'bar'}});
    })
});
```
The node script will print the same headers for both users:
```
Basic Zm9vOmZvbw==
Basic Zm9vOmZvbw==
```

If you change the port in the node script and the Cypress config to e.g. `8080`, the node script will print different headers for the to users (as it should):
```
Basic Zm9vOmZvbw==
Basic YmFyOmJhcg==
```
Furthermore if no base url is configured and the full address is used directly in the test, it seems to work correctly also for port `80`.

**Comment by ryan-snyder** [comment]
@schmist Sorry for not taking a look at this. I'll try to look tonight and see and let you know

**Comment by schmist** [comment]
@jennifer-shehane @ryan-snyder are there any update on this? Right now we are relying on a very ugly workaround involving an SSH tunnel, just to change the port Cypress runs against (since we cannot change the port of the target system).

**Comment by schmist** [comment]
@jennifer-shehane do you still need more information? any updates?

**Comment by schmist** [comment]
This seems to be fixed since (at least) version 3.8.3.

**Comment by jennifer-shehane** [comment]
Closing as resolved. 

If you're experiencing a bug similar to this in Cypress, please [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a fully reproducible example that we can run. There may be a specific edge case with the issue that we need more detail to fix.
